rct_optimizations/include/rct_optimizations/covariance_analysis.h

```diff
@@ -30,40 +38,48 @@ Eigen::MatrixXd computeCorrelationsFromCovariance(const Eigen::MatrixXd& covaria
 
 /**
  * @brief Compute all covariance results for a Ceres optimization problem. Labels results with generic names.
+ * @details This excludes all parameters set to be constant during the solve
  * @param problem The Ceres problem (after optimization).
  * @param options ceres::Covariance::Options to use when calculating covariance.
  * @return CovarianceResult for the problem.
  */
 CovarianceResult computeCovariance(ceres::Problem &problem,
+                                   const std::map<const double *, std::vector<int> >& param_masks = std::map<const double *, std::vector<int>>(),
                                    const ceres::Covariance::Options& options = DefaultCovarianceOptions());
 
 /**
  * @brief Compute covariance results for the specified parameter blocks in a Ceres optimization problem. Labels results with generic names.
+ * @details This excludes all parameters set to be constant during the solve
  * @param problem The Ceres problem (after optimization).
  * @param parameter_blocks Specific parameter blocks to compute covariance between.
  * @param options ceres::Covariance::Options to use when calculating covariance.
  * @return CovarianceResult for the problem.
  */
 CovarianceResult computeCovariance(ceres::Problem &problem,
                                    const std::vector<const double *>& parameter_blocks,
+                                   const std::map<const double *, std::vector<int> >& param_masks = std::map<const double *, std::vector<int>>(),
                                    const ceres::Covariance::Options& options = DefaultCovarianceOptions());
 
 /**
  * @brief Compute all covariance results for a Ceres optimization problem and label them with the provided names.
+ * @details This excludes all parameters set to be constant during the solve
  * @param problem The Ceres problem (after optimization).
  * @param parameter_names Labels for all optimization parameters in the problem.
  * @param options ceres::Covariance::Options to use when calculating covariance.
  * @return CovarianceResult for the problem.
  */
 CovarianceResult computeCovariance(ceres::Problem &problem,
-                                   const std::vector<std::vector<std::string>>& parameter_names,
+                                   const std::map<const double *, std::vector<std::string> > &param_names,
+                                   const std::map<const double *, std::vector<int> >& param_masks = std::map<const double *, std::vector<int>>(),
                                    const ceres::Covariance::Options& options = DefaultCovarianceOptions());
 
 /**
  * @brief Compute covariance results for specified parameter blocks in a Ceres optimization problem and label them with the provided names.
+ * @details This excludes all parameters set to be constant during the solve
```

Comment:
This is a valid comment. This does occur regardless of the mask now.

Reply:
Okay; I assume this is because we are checking `isParameterBlockConstant` in the function?

Reply:
Yea, not sure if there is a better way to handle this. When creating the problem we use the mask data and if all parameters in a group are masked it sets it to constant. Though I could possible remove this check and solely rely on the mask data. @marip8 What do you thing?

Reply:
I'm inclined to say that we remove this check and let the user explicitly provide the mask. The covariance between a variable and a constant is 0, so worst case scenario is that you'll see a bunch of zeros if you forget to mask a constant block. 

Following the idea of autmatically removing constant parameters from the covariance: another thing that makes this tricky is that individual parameters can be held constant using a `SubsetParameterization`. We could also automatically remove those from the covariance matrix (without specifiying a mask) by calling `GetCovarianceMatrixInTangentSpace` instead of `GetCovarianceMatrix`. At this point however, the size/indices of your covariance matrix no longer correspond to the parameter block labels and indices. This would also be made worse if any other local parameterizations were used (e.g., the local parameterization that represents quaternions with 3 values instead of 4).

TLDR: let's get the full covariance matrix (like we're already doing), not automatically remove constant blocks, and rely explicitly on the masks to show the user the covariances they asked to see

Reply:
I just added a commit to only use the mask info.